### PR TITLE
python312Packages.great-tables: fix on darwin

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

One test on darwin was failing with:
```
>       self.socket.bind(self.server_address)
E       PermissionError: [Errno 1] Operation not permitted
/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/lib/python3.12/socketserver.py:473: PermissionError
```

Adding `__darwinAllowLocalNetworking` fixed it.

- **python312Packages.great-tables: cleaning**
- **python312Packages.great-tables: allow local networking on darwin to fix a failing test**

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 341873` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.great-tables</li>
    <li>python311Packages.great-tables.dist</li>
    <li>python312Packages.great-tables</li>
    <li>python312Packages.great-tables.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 341873` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.great-tables</li>
    <li>python311Packages.great-tables.dist</li>
    <li>python312Packages.great-tables</li>
    <li>python312Packages.great-tables.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 341873` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 341873` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/development/python-modules/great-tables/default.nix [hunk 1]
```diff
@@ -28,8 +33,6 @@ buildPythonPackage rec {
   version = "0.11.0";
   pyproject = true;
 
-  disabled = pythonOlder "3.9";
```
Comment (pbsds) [inline pkgs/development/python-modules/great-tables/default.nix:31]: why do you remove these? We still carry older python an pypy versions even though hydra doesn't build them anymore

Comment (GaetanLepage) [reply pkgs/development/python-modules/great-tables/default.nix:31]: Other maintainers argued that keeping the `disabled` metadata when it is older than the python version we officially support was not that useful.
I also find it annoying to always check upstream information to update them. So, I have been removing them lately.

Comment (pbsds) [reply pkgs/development/python-modules/great-tables/default.nix:31]: Personally i prefer keeping them around, as they can save you a lot of time if you suddenly decide to build `python38Packages` for some reason. Perhaps some tooling to test these would be called for?

Comment (pbsds) [reply pkgs/development/python-modules/great-tables/default.nix:31]: (not blocking)

**Pull request merged**